compiler/test/fail_compilation/betterc2.d

```diff
@@ -0,0 +1,17 @@
+/* REQUIRED_ARGS: -betterC
+ * TEST_OUTPUT:
+---
+fail_compilation/betterc2.d(16): Error: `TypeInfo` cannot be used with -betterC
```

Comment:
From what I can tell, the `TypeInfo` would be generated only in expressions, which could be used and allowed in CTFE. It's impossible to limit a declaration to non-CTFE usage because `if (__ctfe)` doesn't exist at compile-time, so for the purpose of not limiting CTFE in betterC code I would only point to the expression for the errors.

Reply:
I made this change in the compiler to make the unittests work in betterC for this file: 

https://github.com/libmir/mir-core/blob/master/source/mir/algebraic.d

The compiler complained about `TypeInfo cannot be used in betterC` while pointing at `Variant` as the problem. I added some logging to DMD and narrowed it down to an associative array issue.

Can you spot the associative array in this file?